Update Copilot example (#46031)
dotnet/docs: docs/standard/serialization/system-text-json/customize-properties.md

:::code language="csharp" source="snippets/how-to-6-0/csharp/PropertyOrder.cs":::

> ##  Use GitHub Copilot to customize how property names are serialized

Reviewer: ```suggestion
## Use GitHub Copilot to customize how property names are serialized
```

> You can use GitHub Copilot with `System.Text.Json` attributes to apply patterns of changes to how your code serializes without having to explicitly specify the change in each one.

Reviewer: ```suggestion
You can prompt GitHub Copilot to use <xref:System.Text.Json> attributes to apply patterns of changes to how your code serializes.
```

Suppose your class declaration has properties that follow `PascalCasing`, and the JSON standard for your project is `snake_casing`. You can use AI to add the necessary `JsonPropertyName` attributes to every property in your class. You can use Copilot to make these changes with a chat prompt like this:

```copilot-prompt
Update #ClassName:
when the property name contains more than one word,

> change the serialized property name to use underscores between words.

Reviewer: Can we just ask it to convert to snake casing instead of having to spell it out?

Author: Will need to test out any prompt changes. To get this out quickly, ok to do this in a different iteration?

> ## Use GitHub Copilot to customize property names and order

Reviewer 2: ```suggestion
## Use GitHub Copilot to customize how property names are serialized
```

## Use GitHub Copilot to customize property names and order

You can use GitHub Copilot with `System.Text.Json` attributes to apply patterns of changes to how your code serializes without having to explicitly specify the change in each one.

Suppose your class declaration has properties that follow `PascalCasing`, and the JSON standard for your project is `snake_casing`. You can use AI to add the necessary `JsonPropertyName` attributes to every property in your class. You can use Copilot Chat to make these changes with a prompt like this (this prompt works best with the Claude 3.5 Sonnet model).

```copilot-prompt
Update #ClassName:
when the property name contains more than one word,
change the serialized property name to use underscores between words.
Serialize the properties in reverse alphabetical order,

> without changing the order in which the properties are listed.

Reviewer 2: ```suggestion
without changing the order in which the properties are listed. Use built-in serialization attributes.
```

## Use GitHub Copilot to customize property names and order

You can use GitHub Copilot with `System.JSON.Text` attributes to apply patterns of changes to how your code serializes without having to explicitly specify the change in each one.

> Suppose you want to change how properties are serialized—formatting property names with underscores between words and ordering them in reverse alphabetical order, while keeping the original property name. You can use Copilot Chat to make these changes with a prompt like this (this prompt works best with the Claude 3.5 Sonnet model).

Reviewer 3: I'd make it more clear why a developer might do this, and how AI is a great tool for this kind of task:

```suggestion
Suppose your class declaration has properties that follow `PascalCasing`, and the JSON standard for your project is `snake_casing`. You can use AI to add the necessary `JsonPropertyName` attributes to every property in your class. You can use Copilot Chat to make these changes with a prompt like this (this prompt works best with the Claude 3.5 Sonnet model).
```

> ```copilot-prompt

Reviewer 4: I don't like that there are two example prompts. It's hard to determine which one I should use, it might easier to just show one example. Ideally, a single prompt for a specific scenario, perhaps the one that is contextual to a value-add as it relates to what the prompt is trying to achieve for this task.

Author: Thanks @IEvangelist! In this case, which prompt do you suggest we use?

> ```copilot-prompt

Reviewer 4: Since the `copilot-prompts` slug doesn't automatically apply a wrapped text, I'd suggest manually adding a newline ever 84 character to avoid horizontal scrolling. I'm never going to blindly copy a prompt; it should be shown in full view without the need for scrolling.

> Update class <your class name>: when the property name contains more than one word, change the serialization name for that property to have underscores between each word.

Reviewer 4: I think the official syntax for referencing files is `#file:<file-name>`.

Author: Its "#className"

Serialize the properties in reverse alphabetical order without changing the order in which the properties are listed.
```

Here's a more complete version of the example that includes a simple class.   
 
```copilot-prompt 
Take this C# class: 
public class WeatherForecast    
{                  

> public DateTime Date { get; set; }         
> public int TemperatureC { get; set; }                 
> public int TemperatureF { get; set; }                
> public string? Summary { get; set; }                  
> public int WindSpeed { get; set; }     

Reviewer 4: Why is the code not formatted correctly? I always format things correctly, even in prompts.

```suggestion
    public DateTime Date { get; set; }
    public int TemperatureC { get; set; }
    public int TemperatureF { get; set; }
    public string? Summary { get; set; }
    public int WindSpeed { get; set; }
```

Author: Thanks. Rushed PR for the discussion.

}
When the property name contains more than one word, change the serialization name for that property to have underscores between each word.
Serialize the properties in reverse alphabetical order without changing the order in which the properties are listed.
Use built-in serialization attributes.
```

> Copilot is powered by AI, so surprises and mistakes are possible. For more information, see [Copilot FAQs](https://aka.ms/copilot-general-use-faqs).

Reviewer 4: Above we call it GitHub Copilot, now it's just Copilot, is that desired? There are many Copilots...
